src/scripts/main.js

```diff
@@ -1,5 +1,28 @@
-'use strict';
+"use strict";
 
-document.addEventListener('click', e => {
-  // write code here
+document.addEventListener("click", (e) => {
+  const spider = document.querySelector(".spider");
+  const wall = document.querySelector(".wall");
+
+  const clickX = e.clientX;
+  const clickY = e.clientY;
+  const wallRect = wall.getBoundingClientRect();
+
+  const spiderX = Math.max(
+    wallRect.left,
+    Math.min(
+      wallRect.right - spider.clientWidth,
```

Comment:
No check if the spider has clientWidth attribute. This may cause errors if the spider element does not have a clientWidth attribute.